index.js

```diff
@@ -24,47 +25,21 @@ function router (controller, routes, options) {
   }
 
   function getUrl (route, input) {
-    var url = route;
-
-    var params = route.match(/:.[^\/]*/g);
-
-    if (params) {
-      // If called from a url change, add params and query to input
-      if (input.route && input.route.params) {
-        input = params.reduce(function (input, param) {
-          var key = param.substr(1, param.length);
-          input[key] = input.route.params[key];
-          return input;
-        }, input);
-      }
-
-      // Create url based on direct signal input or
-      // params passed from addressbar
-      url = params.reduce(function (url, param) {
-        var key = param.substr(1, param.length);
-        if (!(key in input)) {
-          throw new Error('Cerebral router - The signal "' + routes[route] + '" is bound to "' + route + '" route, but required param "' + key + '" wasn\'t provided.');
-        }
-        return url.replace(param, input[key] || '');
-      }, url);
-
-      // Check resulted url still matches given route
-      var urlMatched = false;
-      var checkRoute = {};
-
-      checkRoute[route] = function () {
-        urlMatched = true;
-      };
-
-      urlMapper(url, checkRoute);
-
-      if (!urlMatched) {
-        throw new Error('Cerebral router - Computed url for signal "' + routes[route] +'" can\'t match given route "' + route + '".\n' +
-                        'Check required params provided to signal is not falsy.');
-      }
+    var params = pathToRegexp(route).keys;
+
+    // If called from a url change, add params and query to input
+    if (input.route && input.route.params) {
+      input = params.reduce(function (input, param) {
+        input[param.name] = input.route.params[param.name];
+        return input;
+      }, input);
     }
 
-    url = url === '*' ? location.pathname : url;
+    // Create url based on direct signal input or
+    // params passed from addressbar
+    var url = pathToRegexp.compile(route)(input);
+
+    url = route === '*' ? location.pathname : url;
```

Comment:
we should deprecate '*' route definition. Right way to define catch-all route is '/*'